gammapy/makers/safe.py

```diff
@@ -40,6 +40,12 @@ class SafeMaskMaker(Maker):
         the `aeff_percent` or `bias_percent` are computed.
     offset_max : str or `~astropy.units.Quantity`
         Maximum offset cut.
+    n_95percentile : float
+        If not None the `bkg-clip` method will mask the values larger than 
+        `n_95percentile` times the 95% percentile value.
+        Default is None then only non-finite values are masked.
+    force_clipping : bool
```

Comment:
I guess this is needed to solve #3064? I can see the need for this, but I'd prefer to solve it in a more general way. Such that e.g. `dataset.to_masked()` handles the `NaN` values correctly. Maybe remove from here and we add an option like `dataset.to_masked(nan_to_num=0)`, which passed down to `Map.stack(, weights=, nan_to_num=0)`?

Reply:
in a separate PR ?

Reply:
Yes, please!